odl/solvers/functional/default_functionals.py

```diff
@@ -223,6 +230,9 @@ def __init__(self, vfspace, exponent=None):
             0 and 1 are currently not supported due to numerical
             instability. Infinity gives the supremum norm.
             Default: ``vfspace.exponent``, usually 2.
+        range : `Field`, optional
+            Range of the functional. The default `None` will be treated as
+            `vfspace.field`.
```

Comment:
We need to decide on a default value here. Either we go for `RealNumbers` for all functionals that can only return real values anyway, or we go for `domain.field`. I'm personally not quite sure which convention is the best, but I guess the msot restrictive would be best, which indicates that we should go for `RealNumbers` whenever the functional is purely real valued.